relay/abi.go

```diff
@@ -466,6 +489,43 @@ func handleLog(vLog *types.Log, w *wallet.Wallet, q *uploader.Queue, chainID uin
 
 	log.Println(res.RawLog)
 	log.Println(res.TxHash)
+
+	// Callback on EVM chain
+	account, err := wEth.Derive(hdwallet.MustParseDerivationPath("m/44'/60'/0'/0/0"), false)
+	if err != nil {
+		log.Printf("Failed to derive account: %v", err)
+		return
+	}
+
+	// Generate privkey and command
+	privKey, err := wEth.PrivateKeyHex(account)
+	if err != nil {
+		log.Printf("Failed to generate privkey: %v", err)
+		return
+	}
+
+	// Generate message ID
+	messageId := messageType + evmAddress + strconv.FormatUint(vLog.BlockNumber, 10)
+
+	// Actually execute command
+	cmdArgs := []string{"send", "--rpc-url", RPC, "--private-key", privKey, vLog.Address.Hex(), "finishMessage(string)", messageId}
+	cmd := exec.Command("cast", cmdArgs...)
```

Comment:
is there a go library to do this without needing the cast command to be installed?

Reply:
as discussed on slack, the only option seems to be geth which is too low level